### PR TITLE
Feature/otp enhancements

### PR DESCRIPTION
**Pull request by ScMofeoluwa**
This pull request introduces two key functionalities to the SDK:

- **Verify OTP**: Adds the capability to verify an OTP code against the stored data.
- **Get OTP**: Implements the method for retrieving an OTP code, intended for future use.

**Note**: The 'Get OTP' functionality is currently untested due to an existing issue with the API. I made a complaint regarding this, and I plan to update the pull request with tests once the issue is resolved.

### CODE REVIEW

File: go.mod
Comment (marvinhosea) [inline go.mod:1]: @ScMofeoluwa how about we maintain go version to ```go 1.21```

File: lib/http.go [hunk 1]
```diff
@@ -37,7 +37,24 @@ func (c *APIClient) Post(endpoint string, payload interface{}) (*http.Response,
 	}
 
 	req.Header.Set("Content-Type", "application/json")
-	req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", c.APIKey))
+	req.Header.Set("x-api-key", c.APIKey)
```
Comment (marvinhosea) [inline lib/http.go:40]: I am not sure if this is the X in x-api-key is capital in the documentation. If it doest affect anything then just ignore it.

**Review by marvinhosea — APPROVED** [review]
LGTM apart from minor observations, have a look  at the comment

File: fastotp.go [hunk 2]
```diff
@@ -23,17 +22,19 @@ type ErrorResponse struct {
 	Errors  map[string][]string `json:"errors"`
 	Message string              `json:"message"`
 }
+
 type OTP struct {
 	CreatedAt       time.Time       `json:"created_at"`
-	DeliveryDetails DeliveryDetails `json:"delivery_details"`
-	DeliveryMethods []string        `json:"delivery_methods"`
```
Comment (CeoFred) [inline fastotp.go:29]: why removing this? reference the documentation https://docs.fastotp.co/api-reference/endpoint/generate-otp

File: fastotp.go [hunk 3]
```diff
@@ -23,17 +22,19 @@ type ErrorResponse struct {
...
 	ExpiresAt       time.Time       `json:"expires_at"`
+	UpdatedAt       time.Time       `json:"updated_at"`
+	DeliveryDetails DeliveryDetails `json:"delivery_details"`
 	ID              string          `json:"id"`
 	Identifier      string          `json:"identifier"`
 	Status          string          `json:"status"`
 	Type            string          `json:"type"`
-	UpdatedAt       time.Time       `json:"updated_at"`
```
Comment (CeoFred) [inline fastotp.go:35]: don't remove this it's part of the otp response. check docs https://docs.fastotp.co/api-reference/endpoint/generate-otp

File: fastotp.go [hunk 4]
```diff
@@ -83,7 +88,64 @@ func (f *FastOtp) GenerateOTP(payload GenerateOTPPayload) (*OTP, error) {
 		return nil, err
 	}
 
-	fmt.Println(otpResponse, "<<< ID")
+	return &otpResponse.OTP, nil
+}
+
+func (f *FastOtp) ValidateOtp(payload ValidateOTPPayload) (*OTP, error) {
```
Comment (CeoFred) [inline fastotp.go]: use `ValidateOTP` instead as function name

File: go.mod
Comment (CeoFred) [reply go.mod:1]: yeas, please maintain the go version. thank you.

File: lib/http.go (hunk 1 above)
Comment (CeoFred) [reply lib/http.go:40]: thaks for this .. the header according to the docs is `x-api-key` for authorization

**Review by CeoFred — CHANGES_REQUESTED** [review]

File: fastotp.go (hunk 2 above)
Comment (ScMofeoluwa) [reply fastotp.go:29]: I didn't actually remove anything. I reordered struct for reduced memory usage as per gopls recommendations cc: struct alignment

File: fastotp.go (hunk 3 above)
Comment (ScMofeoluwa) [reply fastotp.go:35]: same struct alignment as mentioned above

File: go.mod
Comment (ScMofeoluwa) [reply go.mod:1]: I'd do just that

File: fastotp.go (hunk 4 above)
Comment (ScMofeoluwa) [reply fastotp.go]: Yessir! 🫡

File: fastotp.go (hunk 2 above)
Comment (CeoFred) [reply fastotp.go:29]: oh. yea.. missed that. nice one

File: fastotp.go (hunk 3 above)
Comment (CeoFred) [reply fastotp.go:35]: yeah. sorry about that.

**Review by CeoFred — APPROVED** [review]
LGTM!

**Comment by CeoFred** [comment]
Please guys, send your twitter usernames. @ScMofeoluwa  @marvinhosea

**Comment by ScMofeoluwa** [comment]
Yessir! 🫡... Btw, my twitter handle is scmofeoluwa

**Pull request merged**